private/buf/cmd/buf/command/push/push.go

```diff
@@ -132,6 +142,29 @@ func (f *flags) Bind(flagSet *pflag.FlagSet) {
 		"",
 		"The URL for viewing the source code of the pushed modules (e.g. the specific commit in source control).",
 	)
+	flagSet.BoolVar(
+		&f.GitMetadata,
+		gitMetadataFlagName,
+		false,
+		fmt.Sprintf(
+			`Uses the Git source control state to set flag values. If this flag is set, we will use the following values for your flags:
+
+	--%s to <git remote URL>/<repository name>/<route>/<checked out commit sha> (e.g. https://github.com/acme/weather/commit/ffac537e6cbbf934b08745a378932722df287a53)
+	--%s for each Git tag and branch pointing to the currently checked out commit
+	--%s to the Git default branch (e.g. main) - this is only in effect if --%s is also set
+
+The source control URL and default branch is based on the Git remote, %q, and requires you to have this remote.
+This flag is only compatible with checkouts of Git source repositories.
+This flag does not allow you to set any of the following flags yourself: --%s, --%s.`,
```

Comment:
Do we need to mention anything about how any additional `--label` parameters will be added in addition to the default labels detected from Git?

Reply:
Hmm, it might not be a bad idea to do so -- the more information the better. I'll add something here for that :)